src/sequence/mod.rs

```diff
@@ -6,6 +6,34 @@ mod macros;
 use internal::IResult;
 use error::ParseError;
 
+/// Gets an object from the first parser,
+/// then gets another object from the second parser.
+/// # Arguments
+/// * `first` The first parser to apply.
+/// * `second` The second parser to apply.
+/// ```rust
+/// # #[macro_use] extern crate nom;
+/// # use nom::{Err, error::ErrorKind, Needed};
+/// # use nom::Needed::Size;
+/// use nom::sequence::pair;
+/// use nom::bytes::complete::tag;
+/// # fn main() {
+/// let first_parser = |s: &'static str| {
+///   tag::<_, _, (_, ErrorKind)>("abc")(s)
+/// };
+/// let second_parser = |s: &'static str| {
+///   tag::<_, _, (_, ErrorKind)>("efg")(s)
+/// };
+/// let parser = |s: &'static str| {
+///   pair::<_, _, _, (_, ErrorKind), _, _>(first_parser, second_parser)(s)
```

Comment:
maybe write the function like this instead:
```rust
fn parser(s: &str) -> IResult<&str, (&str, &str)> {
  pair(first_parser, second_parser)(s)
}
```
All those generic parameters will scare users a bit